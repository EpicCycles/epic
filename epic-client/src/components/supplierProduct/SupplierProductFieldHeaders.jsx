import React, {Fragment} from "react";
import {supplierProductFields} from "../app/model/helpers/fields";
import * as PropTypes from "prop-types";
import {fixedHeaderClassname} from "../app/model/helpers/display";
import ErrorHeader from "../app/model/ErrorHeader";

const SupplierProductFieldHeaders = (props) => {
    return <Fragment>
        {supplierProductFields.map(field => {
            return <div
                className={`grid-item--header ${props.className} ${fixedHeaderClassname(props.lockFirstColumn)}`}
                key={`partHead${field.fieldName}`}
            >
                {field.header}
            </div>;
        })}
        <ErrorHeader/>
    </Fragment>;
};
SupplierProductFieldHeaders.defaultProps = {
    lockFirstColumn: false,
    showErrors: false,
    className: "",
};
SupplierProductFieldHeaders.propTypes = {
    lockFirstColumn: PropTypes.bool,
    showErrors: PropTypes.bool,
    className: PropTypes.string,
};
export default SupplierProductFieldHeaders;