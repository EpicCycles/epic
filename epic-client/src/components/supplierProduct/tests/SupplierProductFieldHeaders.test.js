import React from 'react';
import SupplierProductFieldHeaders from "../SupplierProductFieldHeaders";

test('should display headers', () => {
    const component = shallow(<SupplierProductFieldHeaders/>);
    expect(component).toMatchSnapshot();
});
test('should display headers with locking, error and an extra class', () => {
    const component = shallow(<SupplierProductFieldHeaders
        lockFirstColumn={true}
        className={"pink"}
        showErrors={true}
    />);
    expect(component).toMatchSnapshot();
});