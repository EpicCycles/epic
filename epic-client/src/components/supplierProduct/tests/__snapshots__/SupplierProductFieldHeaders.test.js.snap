// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display headers 1`] = `
<React.Fragment>
  <div
    className="grid-item--header  "
    key="partHeadsupplier"
  >
    Supplier
  </div>
  <div
    className="grid-item--header  "
    key="partHeadpart"
  >
    Part
  </div>
  <div
    className="grid-item--header  "
    key="partHeadproduct_code"
  >
    Product Code
  </div>
  <div
    className="grid-item--header  "
    key="partHeadfitted_price"
  >
    Fitted Price
  </div>
  <div
    className="grid-item--header  "
    key="partHeadticket_price"
  >
    Ticket Price
  </div>
  <div
    className="grid-item--header  "
    key="partHeadrrp"
  >
    RRP
  </div>
  <div
    className="grid-item--header  "
    key="partHeadtrade_price"
  >
    Trade Price
  </div>
  <div
    className="grid-item--header  "
    key="partHeadclub_price"
  >
    Club Price
  </div>
  <div
    className="grid-item--header  "
    key="partHeadcheck_date"
  >
    Date Checked
  </div>
  <ErrorHeader
    className=""
    lockedColumn={false}
  />
</React.Fragment>
`;

exports[`should display headers with locking, error and an extra class 1`] = `
<React.Fragment>
  <div
    className="grid-item--header pink grid-header--fixed-left"
    key="partHeadsupplier"
  >
    Supplier
  </div>
  <div
    className="grid-item--header pink grid-header--fixed-left"
    key="partHeadpart"
  >
    Part
  </div>
  <div
    className="grid-item--header pink grid-header--fixed-left"
    key="partHeadproduct_code"
  >
    Product Code
  </div>
  <div
    className="grid-item--header pink grid-header--fixed-left"
    key="partHeadfitted_price"
  >
    Fitted Price
  </div>
  <div
    className="grid-item--header pink grid-header--fixed-left"
    key="partHeadticket_price"
  >
    Ticket Price
  </div>
  <div
    className="grid-item--header pink grid-header--fixed-left"
    key="partHeadrrp"
  >
    RRP
  </div>
  <div
    className="grid-item--header pink grid-header--fixed-left"
    key="partHeadtrade_price"
  >
    Trade Price
  </div>
  <div
    className="grid-item--header pink grid-header--fixed-left"
    key="partHeadclub_price"
  >
    Club Price
  </div>
  <div
    className="grid-item--header pink grid-header--fixed-left"
    key="partHeadcheck_date"
  >
    Date Checked
  </div>
  <ErrorHeader
    className=""
    lockedColumn={false}
  />
</React.Fragment>
`;
